src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -193,23 +193,39 @@ public IList<IProvider> Children
     [Serializable]
     public class ProviderCategoryConfiguration
     {
-        public string Name { get; set; }
+        public ProviderCategoryConfiguration()
+        {
+            Providers = new Dictionary<string, IProviderConfiguration>();
+        }
 
-        public IDictionary<string, IProviderConfiguration> Providers { get; set; }
+        internal ProviderCategoryConfiguration(string name) : this()
```

Comment:
No. It's an internal ctor. Parameters, should be checked only on boundaries, public surface.